flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/sink/WritableMetadata.java

```diff
@@ -83,4 +84,38 @@ public static TimestampMetadata of(List<String> metadataKeys, DataType physicalD
                     pos + physicalDataType.getLogicalType().getChildren().size());
         }
     }
+
+    /** Time-to-live metadata for HBase. */
+    public static class TimeToLiveMetadata extends WritableMetadata<Long> {
+
+        public static final String KEY = "ttl";
+        public static final DataType DATA_TYPE = DataTypes.BIGINT().nullable();
+
+        private final int pos;
+
+        public TimeToLiveMetadata(int pos) {
+            this.pos = pos;
+        }
+
+        @Override
+        public Long read(RowData row) {
+            if (pos < 0) {
+                return null;
+            }
+            if (row.isNullAt(pos)) {
+                throw new IllegalArgumentException(
+                        String.format("Writable metadata '%s' can not accept null value", KEY));
+            }
```

Comment:
Now this `null` check is basically duplicated and probably will be required for any new metadata type we might have in the future. I think we could separate this into a `private static` method in `WritableMetadata`, something like `validateNotNull(RowData row, int pos, String key)`.